app/app.go

```diff
@@ -82,20 +85,40 @@ func (a *app) HealthCheck(ctx context.Context) error {
 	return a.store.Ping(ctx)
 }
 
-func (a *app) GetSettings(ctx context.Context) (model.Settings, error) {
-	return a.store.GetSettings(ctx)
+func (a *app) GetIntegrations(ctx context.Context) ([]model.Integration, error) {
+	return a.store.GetIntegrations(ctx)
 }
 
-func (a *app) SetSettings(ctx context.Context, settings model.Settings) error {
-	return a.store.SetSettings(ctx, settings)
+func (a *app) GetIntegrationById(ctx context.Context, deviceID string) (model.Integration, error) {
+	deviceId, err := uuid.Parse(deviceID)
+	if err != nil {
+		return model.Integration{}, errors.Wrap(err, "failed to parse deviceID")
+	}
```

Comment:
Parameter parsing/validation should be done in the `api/http` package. The call signature is also mixing device and integration IDs.
```suggestion
func (a *app) GetIntegrationById(ctx context.Context, id uuid.UUID) (model.Integration, error) {
```